### DISCUSSION TITLE
Named arguments with variable length Objects

### DISCUSSION THREAD

**Discussion by ktrueda** [junit-team/junit-framework#2740] [Q&A] [answered]
I'd like to use Named arguments ( #2521 ). 

But I couldn't understand how to use it with variable length Objects like [this test case](https://github.com/ktrueda/junit5/blob/9bcbc5b8c39d7083175e470faca777759b9a64de/documentation/src/test/java/example/ParameterizedTestDemo.java#L202).

I tried this code 
```java
@DisplayName("A parameterized test with named arguments")
@ParameterizedTest(name = "{index}: {0}")
@MethodSource("namedArguments")
void testWithNamedArguments(File file, String str) {
}

static Stream<Arguments> namedArguments() {
return Stream.of(arguments(Named.of("An important file", new Object[] {new File("path1"), "foo"})),
	arguments(Named.of("Another file", new Object[]{new File("path2"), "bar"})));
}
```
and got 
```
org.junit.jupiter.api.extension.ParameterResolutionException: 
Error converting parameter at index 0: No implicit conversion to convert object of type [Ljava.lang.Object; to type java.io.File
```
.

I'd be thankful for any hint.

**Comment by sbrannen** [comment, marked as answer]
Only the file is a "named" argument.

Use the following to achieve your goal.

```java
	static Stream<Arguments> namedArguments() {
		return Stream.of(
			arguments(named("An important file", new File("path1")), "foo"),
			arguments(named("Another file", new File("path2")), "bar")
		);
	}
```

**Reply by ktrueda** [reply]
Thank you for your reply.
Your code works. 

But this is weird a little.
I wanted to name not the file but the arguments( file and string). 

If I can use `Named.of(String name, Object... payload)` , I think it is straight and easy to understand.

**Reply by sbrannen** [reply]
>  I wanted to name not the file but the arguments( file and string).

The solution I provided names the `File` argument.

If you want to name the `String` argument some custom name as well, you will have to wrap that in `Named` as well.

> If I can use Named.of(String name, Object... payload) , I think it is straight and easy to understand.

`Named` can only be used to wrap a single payload (argument in this scenario).

**Reply by ktrueda** [reply]
Thanks. I understand Named is not what I wanted.

This is good for me.
```java
@ParameterizedTest(name = "{0}")
@MethodSource("namedArguments")
void testWithNamedArguments(String testCaseName, File file, String str) {
}

static Stream<Arguments> namedArguments() {
return Stream.of(
	arguments(
		"path1 + foo",
		new File("path1"),
		"foo"
	),
	arguments(
		"path2 + bar",
		new File("path2"),
		"bar"
	)
);
}
```
